crates/component-macro/src/bindgen.rs

```diff
@@ -20,13 +21,22 @@ pub struct Config {
 }
 
 pub fn expand(input: &Config) -> Result<TokenStream> {
-    if !cfg!(feature = "async") && input.opts.async_.maybe_async() {
+    if let (CallStyle::Async | CallStyle::Concurrent, false) =
+        (input.opts.call_style(), cfg!(feature = "async"))
```

Comment:
TIL this is a neat idiom for lack of `if let foo = bar && some_other_cond { ... }`